opendse-io/src/main/java/net/sf/opendse/io/TGFFReader.java

```diff
@@ -0,0 +1,443 @@
+package net.sf.opendse.io;
+
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import net.sf.opendse.model.Application;
+import net.sf.opendse.model.Communication;
+import net.sf.opendse.model.Dependency;
+import net.sf.opendse.model.Link;
+import net.sf.opendse.model.Mapping;
+import net.sf.opendse.model.Mappings;
+import net.sf.opendse.model.Resource;
+import net.sf.opendse.model.ResourceTypes;
+import net.sf.opendse.model.TGFFSpecification;
+import net.sf.opendse.model.Task;
+
+
+/**
+ * The {@code TgffReader} imports an {@code Application}, {@code Mappings} and a set of {@code Resource} types from a .tgff-file, 
+ * as generated by Task Graphs For Free (TGFF).
+ * 
+ * In particular, it serves to import benchmark task graphs, processor types and corresponding mapping possibilities from the 
+ * Embedded Systems Synthesis Benchmarks Suite (E3S).
+ * See <a href="http://ziyang.eecs.umich.edu/~dickrp/e3s/">http://ziyang.eecs.umich.edu/~dickrp/e3s/</a>
+ * 
+ * @author Valentina Richthammer 
+ */
+public class TGFFReader {
+	
+	private static final String HYPERPERIOD = "@HYPERPERIOD";
+	private static final String TASK_GRAPH = "@TASK_GRAPH";
+	private static final String COMMUN_QUANT = "@COMMUN_QUANT";
+	private static final String CORE = "@CORE";
+	private static final String CLIENT_PE = "@CLIENT_PE";
+	private static final String SERVER_PE = "@SERVER_PE";
+	private static final String PROC = "@PROC";
+	private static final String LINK = "LINK";  // also for @CLIENT_LINK, @SERVER_LINK, @PRIM_LINK -> no "@"
+	
+	private static final String TASK = "TASK";
+	private static final String ARC = "ARC";
+	
+	private static final String VALID = "valid";
+	private static final String TYPE = "type";
+	
+	private static final String CLOSING = "}";
+	private static final String COMMENT = "#";
+	private static final String AT = "@";
+	private static final String SEPARATOR = "\\s+";
+	protected static final String CONNECTOR = "_";
+	
+	public static final String TGFF_TYPE = "TGFF_TYPE";
+	public static final String PERIOD = "PERIOD"; 
+	public static final String MSG_SIZE = "MSG_SIZE";
+	public static final String RES_ATTRIBUTES = "RES_ATTRIBUTES";
+	public static final String RES_VALUES = "RES_VALUES";
+	public static final String HARD_DEADLINE = "HARD_DEADLINE";
+	public static final String SOFT_DEADLINE = "SOFT_DEADLINE";
+	public static final String WIRE = "@WIRING";
+	
+	
+	protected boolean parsed = false;
+	protected String fileName;
+	protected double hyperperiod;
+	protected String memory;
+
+	protected Application<Task, Dependency> application;
+	protected ResourceTypes<Resource> resourceTypes;
+	protected Mappings<Task, Resource> mappings;

```

Comment:
Please try to reduce the amount of fields. Instead, use respective method signatures (some hints below).